osquery/tables/events/windows/ntfs_journal_events.cpp

```diff
@@ -0,0 +1,373 @@
+    }
+
+    if (access_frns.find(event.parent_ref_number) != access_frns.end()) {
+      access_frns.insert(event.node_ref_number);
+      return true;
+    }
+
+    if (access_paths.find(event.path) != access_paths.end()) {
+      access_frns.insert(event.node_ref_number);
+      return true;
+    }
+
+    return false;
+  }
+}
+
+Row NTFSEventSubscriber::generateRowFromEvent(const NTFSEventRecord& event) {
+  Row row;
+
+  auto action_description_it = kNTFSEventToStringMap.find(event.type);
+  assert(action_description_it != kNTFSEventToStringMap.end());
+
+  row["action"] = TEXT(action_description_it->second);
+  row["old_path"] = TEXT(event.old_path);
+  row["path"] = TEXT(event.path);
+  row["partial"] = INTEGER(event.partial);
+
+  // NOTE(ww): These are emitted in decimal, not hex.
+  // There's no good reason for this, other than that
+  // boost's mp type doesn't handle std::hex and other
+  // ios formatting directives correctly.
+  row["node_ref_number"] = TEXT(event.node_ref_number.str());
+  row["parent_ref_number"] = TEXT(event.parent_ref_number.str());
+
+  {
+    std::stringstream buffer;
+    buffer << event.record_timestamp;
+    row["record_timestamp"] = TEXT(buffer.str());
+
+    buffer.str("");
+    buffer << std::hex << std::setfill('0') << std::setw(16)
+           << event.update_sequence_number;
+    row["record_usn"] = TEXT(buffer.str());
+
+    // NOTE(ww): Maybe comma-separate here? Pipes make it clear
+    // that these are flags, but CSV is easier to parse and is
+    // used by other tables.
+    buffer.str("");
+    bool add_separator = false;
+    for (const auto& p : kWindowsFileAttributeMap) {
+      const auto& bit = p.first;
+      const auto& label = p.second;
+
+      if ((event.attributes & bit) == 0) {
+        continue;
+      }
+
+      if (add_separator) {
+        buffer << " | ";
+      }
+
+      buffer << label;
+      add_separator = true;
+    }
+
+    row["file_attributes"] = TEXT(buffer.str());
+  }
+
+  std::string drive_letter(1, event.drive_letter);
+  row["drive_letter"] = TEXT(drive_letter);
+
+  return row;
+}
+
+NTFSEventSubscriber::NTFSEventSubscriber() {}
+
+NTFSEventSubscriber::~NTFSEventSubscriber() {}
+
+Status NTFSEventSubscriber::init() {
+  return Status(0);
+}
+
+void NTFSEventSubscriber::configure() {
+  // List the categories in the file_accesses array
+  const auto& json = Config::getParser("file_paths")->getData();
+  const auto& json_document = json.doc();
+
+  StringList access_categories;
+  if (json_document.HasMember("file_accesses")) {
+    auto& json_file_accesses = json_document["file_accesses"].GetArray();
+
+    for (const auto& item : json_file_accesses) {
+      access_categories.push_back(item.GetString());
+    }
+  }
+
+  Config::get().files(
+      [this, &json_document, &access_categories](
+          const std::string& category, const std::vector<std::string>& files) {
+        StringList include_path_list = {};
+        for (auto file : files) {
```

Comment:
I am not sure if `auto` here will give you a `const&` (which is what you intend I think). Perhaps `auto& file`?